jbmc/src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -51,11 +51,6 @@ void java_bytecode_languaget::get_language_options(const cmdlinet &cmd)
   throw_assertion_error = cmd.isset("throw-assertion-error");
   threading_support = cmd.isset("java-threading");
 
-  if(cmd.isset("java-max-input-array-length")) // will go away
-  {
-    object_factory_parameters.max_nondet_array_length =
-      safe_string2size_t(cmd.get_value("java-max-input-array-length"));
-  }
```

Comment:
No, this has been a long time hack. The front end is and has been doing that _in addition_ to the language module. Avoiding that would require `options` to be fed into the language module, which is a separate refactoring (and will eventually become obsolete with Daniel's ambitions to get rid of languaget).

Reply:
But we do we need to keep doing it _in addition_ in here? I don't see how the `options` story applies here: it's all being passed via `object_factory_parameters` already?

Reply:
The same options are both used by the `languaget` instance and in goto passes. 
The `object_factory_parameters` in parse_options is only used in goto passes. The `languaget` instance does its own parsing.

Reply:
One could either retrieve the `object_factory_parameters` from a (temporary) `languaget` instance or pass `options` into the `languaget` instance.

Reply:
I don't have a sufficient grasp of the Java front-end to tell what the correct solution is, but surely parsing the same command-line option multiple times is not acceptable.

Reply:
This will definitely require a solution as we will face the same issue when generalising/porting the object_factory to C. And we won't be able to wait until we got rid of `languaget` in this case. (I'm still remembering the incident with https://github.com/diffblue/cbmc/pull/2192, https://github.com/diffblue/cbmc/pull/2157 which still hasn't been resolved after 4 months).

Reply:
I'm happy to dismiss my blocking review on this one if the problem cannot be solved right here. I just want to make sure it does get resolved eventually, with first steps defined. The large-scale refactorings surely aren't the solution...

Reply:
Noted here: https://github.com/diffblue/cbmc/issues/2908